cmd/limactl/prune.go

```diff
@@ -1,30 +1,172 @@
 package main
 
 import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
+	"github.com/lima-vm/lima/pkg/store"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
 
 func newPruneCommand() *cobra.Command {
 	pruneCommand := &cobra.Command{
-		Use:               "prune",
-		Short:             "Prune garbage objects",
-		Args:              WrapArgsError(cobra.NoArgs),
-		RunE:              pruneAction,
-		ValidArgsFunction: cobra.NoFileCompletions,
+		Use:   "prune",
+		Short: "Prune garbage objects",
+		RunE:  pruneAllAction, // keep old behavior
 	}
+
+	pruneCommand.AddCommand(
+		newPruneAllCommand(),
+		newPruneDownloadsCommand(),
+	)
+
 	return pruneCommand
 }
 
-func pruneAction(cmd *cobra.Command, args []string) error {
-	ucd, err := os.UserCacheDir()
+func newPruneAllCommand() *cobra.Command {
+	pruneAllCommand := &cobra.Command{
+		Use:   "all",
+		Short: "Prune all caches",
+		RunE:  pruneAllAction,
+	}
+	return pruneAllCommand
+}
+
+func pruneAllAction(cmd *cobra.Command, args []string) error {
+	cacheDir, err := getCacheDir()
 	if err != nil {
 		return err
 	}
-	cacheDir := filepath.Join(ucd, "lima")
-	logrus.Infof("Pruning %q", cacheDir)
+	logrus.Infof("Pruning everything in %q", cacheDir)
 	return os.RemoveAll(cacheDir)
 }
+
+func newPruneDownloadsCommand() *cobra.Command {
+	pruneDownloadsCommand := &cobra.Command{
+		Use:   "downloads",
+		Short: "Prune unreferenced downloads",
```

Comment:
I'm not sure about the usecase of this.
"Referenced" blobs are copied to `~/.lima/<INSTANCE>/basedisk` and can be removed safely.

Reply:
it's about download speed mostly. I decided to implement this while staying in Medan city in Indonesia because every time I need to spawn another image I should sit and wait for ~30-60min to download 700-1500Mb. So pruning the whole cache after set of experiments was not really good idea.

Though I agree we can move this one CLI level up. I.e. change the default `lima prune` behavior if downloads are the only cached objects we have.

Reply:
Wondering something like `limactl prune --keep="3 days"` may make more sense.

Images that have been used within 3 days will be kept in the cache, regardless to whether it is actually used by any active instance, so that you can avoid downloading when you are going to create an instance from it.

Reply:
Ah, right.. I was thinking about TTL yesterday too.. let's bring this to discussion in #1410